src/libasr/codegen/asr_to_llvm.cpp

```diff
@@ -7901,6 +7901,10 @@ ptr_type[ptr_member] = llvm_utils->get_type_from_ttype_t_util(
         llvm::Type* target_base_type = llvm_utils->get_type_from_ttype_t_util(x.m_type, module.get());
```

Comment:
if you need to need `target_base_type` overriding the Array from over it, why not just do:
```suggestion
        llvm::Type* target_base_type = llvm_utils->get_type_from_ttype_t_util(ASRUtils::type_get_past_array(x.m_type), module.get());
```

that's a cleaner solution as I understand it.